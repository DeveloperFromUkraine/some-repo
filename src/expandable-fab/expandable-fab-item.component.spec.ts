import { ComponentTest } from '../../test/test-bed/component';
import { ExpandableFabItemComponent } from 'index';
import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { MatIconModule } from '@angular/material';
import { By } from '@angular/platform-browser';
import { DebugElement } from '@angular/core/src/debug/debug_node';

describe('ExpandableFabItemComponent', () => {
    let fixture: ComponentFixture<ExpandableFabItemComponent>;
    let component: ExpandableFabItemComponent;
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        await ComponentTest.createTestBed([MatIconModule], [ExpandableFabItemComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(ExpandableFabItemComponent);
        component = fixture.componentInstance;

        fixture.detectChanges();
    });

    it('should set tooltipText', async () => {
        de = fixture.debugElement.query(By.css('.mat-tooltip'));
        ne = de.nativeElement;

        component.tooltipText = 'toolTip';
        await fixture.detectChanges();

        expect(ne.textContent).toEqual(component.tooltipText);
    });

    it('should set icon', async () => {
        de = fixture.debugElement.query(By.css('mat-icon'));
        ne = de.nativeElement;

        component.icon = 'email';
        await fixture.detectChanges();

        expect(ne.textContent).toContain(component.icon);
    });

    it('should set href', async () => {
        de = fixture.debugElement.query(By.css('a'));
        ne = de.nativeElement;

        component.href = 'updatedHref';
        await fixture.detectChanges();

        expect(ne.getAttribute('href')).toBe(component.href);
    });

    it('should emit clicked event', () => {
        de = fixture.debugElement.query(By.css('a'));
        let spy = jest.fn();

        component.clicked.subscribe(spy);
        de.triggerEventHandler('click', null);
        fixture.detectChanges();

        expect(spy).toHaveBeenCalled();
    });

    it('should match snapshot', async () => {
        component.tooltipText = 'toolTip';
        component.icon = 'email';
        component.href = 'updatedHref';
        
        await fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });
});