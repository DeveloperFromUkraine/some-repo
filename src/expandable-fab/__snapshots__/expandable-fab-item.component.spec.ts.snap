// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expandable Fab Item should match snapshot 1`] = `
<ign-expandable-fab-item
  clicked={[Function EventEmitter]}
  href={[Function String]}
  icon={[Function String]}
  tooltipText={[Function String]}
>
  <span
    class="mat-tooltip fab-tooltip"
  >
    toolTip
  </span>
  <a
    href="updatedHref"
    igna11yclick=""
    mat-fab=""
    role="menuitem"
  >
    
    
    <mat-icon
      aria-hidden="true"
      class="mat-icon material-icons"
      role="img"
    >
      
      email
    
    </mat-icon>
    
  
  </a>
</ign-expandable-fab-item>
`;
